### PR TITLE
Add a cvar to display distances

### PR DESCRIPTION
**Pull request by necessarily-equal**
It's a bit hacky / not too practical, but at least it gives a way to estimate distances in-game.

### CODE REVIEW

File: src/cgame/cg_ents.cpp [hunk 1]
```diff
@@ -1373,6 +1373,18 @@ void CG_AddPacketEntities()
 			}
 		}
 	}
+
+	static Cvar::Range<Cvar::Cvar<float>>
```
Comment (bmorel) [inline src/cgame/cg_ents.cpp:1377]: Usually, CVars are extern, not static. They're also defined and declared outside of functions.
Any reason to change that convention?

File: src/cgame/cg_ents.cpp [hunk 2]
```diff
@@ -1373,6 +1373,18 @@ void CG_AddPacketEntities()
...
+		cg_drawDebugDistance("cg_drawDebugDistance", "draw a circle around "
+				"yourself to give an idea of distances. Set this "
+				"to 0 to disable", Cvar::CHEAT, 0.0f, 0.0f,
+				5000.0f);
+	//make an attempt at drawing bounding boxes of selected entity types
```
Comment (bmorel) [inline src/cgame/cg_ents.cpp:1382]: I don't understand. Is this about drawing a circle to estimate distance, or is it about BBoxes?

File: src/cgame/cg_ents.cpp (hunk 1 above)
Comment (necessarily-equal) [reply src/cgame/cg_ents.cpp:1377]: Only real reason is build time

File: src/cgame/cg_ents.cpp (hunk 2 above)
Comment (necessarily-equal) [reply src/cgame/cg_ents.cpp:1382]: Whoops, comment sneaked in. Removing

File: src/cgame/cg_ents.cpp (hunk 1 above)
Comment (bmorel) [reply src/cgame/cg_ents.cpp]: Yes it is uncommon in gamelogic. Only 3 Cvars from thse are not declared at top of file. Duplicated code is also common, yet it's considered a bad thing.

You need the '-n' parameter to grep to know where the cvars are put in the file, -c is only count occurrences.

Comment (necessarily-equal) [reply src/cgame/cg_ents.cpp]: Just moved it at the beginning of the file.

**Comment by bmorel** [comment]
LGTM

**Comment by bmorel** [comment]
The result is very weird.
This actually does not draws a circle, but a sphere.

2022-07-21-114603_1366x768_scrot

It works, but I'm not sure about this PR still.

**Comment by necessarily-equal** [comment]
Yes it does look a bit weird and isn't too practical. But tbh everything works with spheres in-game (when it's not about bounding boxes), and it's still the only way I know to display distances.

**Comment by bmorel** [comment]
No, not everything works with spheres, and its possible to display circles. Look at /navedit command. Also, you didn't fixed the wording.

**Comment by slipher** [comment]
An alternative is to draw a trace from the player's eye and display the distance to the entity the trace hits, similar to `cg_drawEntityInfo` (or maybe even part of it). Of course this isn't perfect as you can't estimate the distance to something which isn't solid. The sphere thing looks even worse though :laughing:

**Pull request merged**